docs/data/data-grid/state/state.md

```diff
@@ -66,40 +66,36 @@ Some selectors are yet to be documented.
 
 ## Save and restore the state
 
-> ⚠️ This feature isn't implemented yet. It's coming.
+The current state of the grid can be exported using `apiRef.current.exportState()`.
+It can then be restored by either passing it to the `initialState` prop or to the `apiRef.current.restoreState()` method.
+
+Watch out for controlled models and their callbacks (`onFilterModelChange` if you use `filterModel` for instance), the grid will call those callbacks when restoring the state.
+But if the callback is not defined or if calling it does not update the prop value, then the restored value will not be applied.
+
+### Restore the state with `initialState`
+
+> ⚠️ If you restore the page using `initialState` before the data is fetched, the grid will automatically move to the 1st page.
+
+{{"demo": "RestoreStateInitialState.js", "bg": "inline", "defaultCodeOpen": false}}
```

Comment:
When I click "Remount the grid" the alert is taking too much space, exceeding the page width. Instead of unmounting and mounting again, we could have a second DataGrid instance and when the button is clicked it transfers the state to the other grid. Unmounting and mounting again works but the user may forget the previous values.

Could you open another PR and delete this demo so it appears as a new file here and we can make suggestions to its code? I guess only renaming won't work.

Reply:
I do a automerge PR where I delete `RestoreStateInitialState` files so that the PR we are one gets the ful file ?

Reply:
Done

Reply:
> we could have a second DataGrid instance and when the button is clicked it transfers the state to the other grid. Unmounting and mounting again works but the user may forget the previous values.

The demo works with an `initialState` so we have to remount the grid in which we inject.
That can be a 2nd grid instead of unmounting the main one.